tests/tool_parsers/test_hy_v4_tool_parser.py

```
@@ -0,0 +1,350 @@
+                "<tool_calls><tool_call>send_message\n"
+                "<arg_key>text</arg_key>"
+                '<arg_value>He said "hello"</arg_value>\n'
+                "<arg_key>path</arg_key>"
+                "<arg_value>C:\\Users\\file</arg_value>\n"
+                "</tool_call></tool_calls>"
+            ),
+            malformed_input_outputs=[
+                # no inner <tool_call> tags -> nothing extracted
+                "<tool_calls>get_weather\n"
+                "<arg_key>city</arg_key><arg_value>Paris</arg_value>\n"
+                "</tool_calls>",
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)